### PR TITLE
Add checker-qual-android artifact

### LINKED ISSUE #2135: Create checker-qual-android artifact
This is a feature request to provide an artifact `checker-qual-android` that is just like `checker-qual` except the annotations have class retention.

Types annotated with runtime annotations are always kept in the main dex by the default Android Gradle plugin. Using the standard Checker Framework annotations can lead to main dex overflows; users of the Checker framework may find themselves unable to build their Android apps.  By contrast, class-retention annotations are stripped out before packaging by all build systems as a convention.

This issue was originally [raised](https://github.com/typetools/checker-framework/pull/2118#issuecomment-414829990) by @kageiit in #2118.

### PR DESCRIPTION
**Pull request by kageiit**
Types annotated with runtime annotations are always kept in the main dex by the default Android Gradle plugin. Using the standard Checker Framework annotations can lead to main dex overflows; users of the Checker framework may find themselves unable to build their Android apps. By contrast, class-retention annotations are stripped out before packaging by all build systems as a convention.

Fixes #2135

### CODE REVIEW

**Comment by kageiit** [comment]
cc @mernst @lazaroclapp

**Comment by lazaroclapp** [comment]
Hey @kageiit, heads up, on running `./gradlew cloneAndBuildDependencies` as part of a fresh from-scratch build (see https://checkerframework.org/manual/#build-source), this fails for me with the error:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/lazaro/checker-framework/checker-qual-android/build.gradle' line: 15

* What went wrong:
Execution failed for task ':checker-qual-android:replaceAnnotations'.
> java.io.FileNotFoundException: /Users/lazaro/checker-framework/checker-qual-android/src/compatQual/java/org/checkerframework/checker/nullness/compatqual/MonotonicNonNullDecl.java (Permission denied)
```

Looking into it now.

**Comment by mernst** [comment]
Thanks for this.  It looks OK to me; I have assigned it to Suzanne for review.

File: checker-qual-android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    from compileCompatQualJava.outputs
+    manifest {
+        attributes('Automatic-Module-Name': "org.checkerframework.checker.qual")
+    }
+}
+
+task compatQualSourcesJar(type: Jar) {
+    description 'Creates compat-qual sources jar.'
+    classifier = 'source'
+    baseName = 'checker-compat-qual-android'
```
Comment (smillst) [inline checker-qual-android/build.gradle]: We are going to stop releasing checker-compat-qual.jar.  Is there any reason to have an Android version?

File: checker-qual-android/build.gradle [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    into file('src')
+    fileMode(0555)
+}
+copySources.dependsOn(":checker-qual:copySources")
+copySources.dependsOn(":checker-qual:copyCompatQual")
+
+task replaceAnnotations {
+    doLast {
+        fileTree(dir: 'src', include: "**/*.java").each {
+            it.text = it.text.replaceAll("RetentionPolicy.RUNTIME", "RetentionPolicy.CLASS")
```
Comment (smillst) [inline checker-qual-android/build.gradle]: Would you please add a comment explaining why `RUNTIME` is preferable to `CLASS` for Android?  

There's text in the issue that you could copy-paste:

>Types annotated with runtime annotations are always kept in the main dex by the default Android Gradle plugin. Using the standard Checker Framework annotations can lead to main dex overflows; users of the Checker framework may find themselves unable to build their Android apps. By contrast, class-retention annotations are stripped out before packaging by all build systems as a convention.

**Comment by mernst** [comment]
I don't see any mention of checker-qual-android in the documentation.  That needs to be added before this can be merged.  Otherwise, users won't know the artifact exists.

**Comment by kageiit** [comment]
> I don't see any mention of checker-qual-android in the documentation. That needs to be added before this can be merged. Otherwise, users won't know the artifact exists.

@mernst where is the best place to add such documentation?

**Comment by mernst** [comment]
I added some text to the manual, in e9a4bb1.

**Comment by mernst** [comment]
I have assigned this back to @smillst because I think it's ready for her review.  However, please let me know if I am mistaken and @kageiit needs to do something more.

**Pull request merged**